xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -3132,8 +3144,52 @@ int CVideoPlayer::GetSubtitle()
   return m_SelectionStreams.IndexOf(STREAM_SUBTITLE, *this);
 }
 
+void CVideoPlayer::UpdateStreamInfos()
+{
+  CSingleLock lock(m_player_status_lock);
+  int streamId;
+  std::string retVal;
+
+  // video
+  streamId = GetVideoStream();
```

Comment:
this accesses SelectionsStreams 3 times without a lock. between two calls the struct may have changed